src/test/java/easytests/core/services/TopicsServiceTest.java

```diff
@@ -3,100 +3,261 @@
+        final List<TopicModelInterface> topicModels = this.getTopicsFixturesModels();
+        final TopicsOptionsInterface topicOptions = mock(TopicsOptionsInterface.class);
+
+        final SubjectModelInterface subjectModel = this.subjectsSupport.getModelFixtureMock(0);
+
+        when(this.topicsMapper.findBySubjectId(subjectModel.getId())).thenReturn(topicEntities);
+        when(topicOptions.withRelations(listCaptor.capture())).thenReturn(topicModels);
+
+        final List<TopicModelInterface> topicFoundedModels =
                 this.topicsService.findBySubject(subjectModel, topicOptions);
 
-        verify(this.topicsMapper).findBySubjectId(subjectId);
-        verify(topicOptions).withRelations(topicsModels);
-        Assert.assertEquals(topicsModels, foundedTopicsModels);
+        this.topicsSupport.assertModelsListEquals(topicModels, listCaptor.getValue());
+        Assert.assertSame(topicModels, topicFoundedModels);
+        this.assertServiceSet(topicOptions);
     }
 
+    @Test
+    public void testSaveCreatesEntity() throws Exception {
+        final ArgumentCaptor<TopicEntity> topicEntityCaptor = ArgumentCaptor.forClass(TopicEntity.class);
+
+        this.topicsService.save(this.topicsSupport.getModelAdditionalMock(0));
+
+        verify(this.topicsMapper, times(1)).insert(topicEntityCaptor.capture());
+        this.topicsSupport.assertEquals(this.topicsSupport.getEntityAdditionalMock(0), topicEntityCaptor.getValue());
+    }
+
+    @Test
+    public void testSaveUpdatesEntityIdOnCreation() throws Exception {
+        final TopicModelInterface topicAdditionalModel = this.topicsSupport.getModelAdditionalMock(0);
+        final Integer updatedId = 10;
+        doAnswer(invocation -> {
+            final TopicEntity topicEntity = invocation.getArgument(0);
+            topicEntity.setId(updatedId);
+            return null;
+        }).when(this.topicsMapper).insert(any(TopicEntity.class));
+
+        this.topicsService.save(topicAdditionalModel);
+
+        verify(topicAdditionalModel, times(1)).setId(updatedId);
+    }
+
+    @Test
+    public void testSaveUpdatesEntity() throws Exception {
+        final ArgumentCaptor<TopicEntity> topicEntityCaptor = ArgumentCaptor.forClass(TopicEntity.class);
+
+        this.topicsService.save(this.topicsSupport.getModelFixtureMock(0));
+
+        verify(this.topicsMapper, times(1)).update(topicEntityCaptor.capture());
+        this.topicsSupport.assertEquals(this.topicsSupport.getModelFixtureMock(0), topicEntityCaptor.getValue());
+    }
+
+    @Test
+    public void testSaveWithOptions() throws Exception {
+        final TopicModelInterface topicModel = this.topicsSupport.getModelFixtureMock(0);
+        final TopicsOptionsInterface topicOptions = mock(TopicsOptionsInterface.class);
+
+        this.topicsService.save(topicModel, topicOptions);
+
+        verify(topicOptions, times(1)).saveWithRelations(topicModel);
+        this.assertServiceSet(topicOptions);
```

Comment:
add verifyNoMoreInteractions

Reply:
@malinink Didn't we want to remove all of them? Why does it needed right there? (and onto deleteTest)

Reply:
@Ielay Because we want to be sure that we pass income model into the option class. And do not put it into mapper in our service on that process.
Because we don't really know _when_ we relly need to seve/delete model (when it is linked with other models).